docs/source/examples/ddd_tutorial.myst

```diff
@@ -0,0 +1,285 @@
+---
+jupytext:
+  text_representation:
+    extension: .myst
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.1
+kernelspec:
+  display_name: Python 3
+  language: python
+  name: python3
+---
+
+# Digital dynamical decoupling (DDD) with Mirror Circuits
+
+In this notebook DDD is applied to improve the success rate of the computation. 
+In DDD, sequences of gates are applied to slack windows, i.e. single-qubit idle windows, in a quantum circuit. 
+Applying such sequences can reduce the coupling between the qubits and the environment, mitigating the effects of noise. 
+For more information on DDD, see the section [DDD section of the user guide](../guide/ddd.myst).
+
+## Setup
+
+We begin by importing the relevant modules and libraries that we will require
+for the rest of this tutorial.
+
+```{code-cell} ipython3
+import functools
+from typing import List, Tuple, Union
+
+# Plotting imports.
+import matplotlib.pyplot as plt
+
+plt.rcParams.update({"font.family": "serif", "font.size": 15})
+%matplotlib inline
+
+# Third-party imports.
+import cirq
+
+import networkx as nx
+import numpy as np
+
+# Mitiq imports.
+from mitiq import benchmarks, ddd
+```
+
+## Define parameters
+
+```{code-cell} ipython3
+# Random seed for circuit generation.
+seed: int = 1
+
+# Total number of shots to use.
+shots: int = 1000
+
+# Qubits to use on the experiment.
+qubits: List[int] = [0, 1, 2]
+
+# Average results over this many trials (circuit instances) at each depth.
+trials: int = 3
+
+# Clifford depths.
+depths: List[int] = [1, 5, 9]
```

Comment:
The type hints shouldn't be needed here since type-checkers can generally infer the type from simple definitions like this.

I think in the future we should decide whether we want our tutorials to have typehints or not. I think there are good arguments for both cases.

Reply:
Copy/paste from previous tutorial. Removing them now.

Reply:
Copy/paste from PEC mirror circuits. Removing them now.